### PR TITLE
Refcount cleanup

### PR DESCRIPTION
**Pull request by jesings**
Correctness fixes and code cleanup for link counts on inodes

### CODE REVIEW

File: src/safeposix/syscalls/net_calls.rs [hunk 1]
```diff
@@ -185,24 +173,46 @@ impl Cage {
...
+                        let mode = (dir.mode | S_FILETYPEFLAGS as u32) & S_IRWXA;
+                        let effective_mode = S_IFSOCK as u32 | mode;
     
-                    let time = interface::timestamp(); //We do a real timestamp now
-                    let newinode = Inode::Socket(SocketInode {
-                        size: 0, uid: DEFAULT_UID, gid: DEFAULT_GID,
-                        mode: effective_mode, linkcount: 1, refcount: 1,
-                        atime: time, ctime: time, mtime: time,
-                    });
+                        let time = interface::timestamp(); //We do a real timestamp now
```
Comment (rennergade) [inline src/safeposix/syscalls/net_calls.rs:184]: weird whitespace in these lines

**Review by rennergade — COMMENTED** [review]
Looks very good. Left a few surface level comments.

File: src/safeposix/syscalls/fs_calls.rs [hunk 2]
```diff
@@ -267,76 +270,66 @@ impl Cage {
...
+
+                    match *inodeobj {
+                        Inode::File(ref mut normalfile_inode_obj) => {
+                            normalfile_inode_obj.linkcount -= 1;
+                        }
+
+                        Inode::CharDev(ref mut chardev_inode_obj) => {
+                            chardev_inode_obj.linkcount -= 1;
+                        }
+
```
Comment (Devansh-Patel) [inline src/safeposix/syscalls/fs_calls.rs:322]: I think this extra newline is not needed.

Comment (jesings) [reply src/safeposix/syscalls/fs_calls.rs:322]: Fixed

**Pull request merged**